### PR TITLE
Lightning GPU v0.25.0 release candidate

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** This branch holds the upcoming changes for the v0.25.0 release of PennyLane Lightning GPU.

**Description of the Change:** Includes bugfixes, and documentation updates for the upcoming release.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -1,12 +1,26 @@
-# Release 0.25.0-dev
+# Release 0.25.0
+
+### New features since last release
+
+* Added support for multi-GPU adjoint observable batching. [(#27)](https://github.com/PennyLaneAI/pennylane-lightning-gpu/pull/27)
```
Comment (mlxd) [inline .github/CHANGELOG.md:5]:
```suggestion
* Added support for multi-GPU adjoint observable batching. [(#27)](https://github.com/PennyLaneAI/pennylane-lightning-gpu/pull/27)

  This new feature allows users to batch their gradients over observables using the `adjoint` method. Assuming multiple GPUs on a host-system, this functionality can be enabled by adding the `batch_obs=True` argument when creating a device, such as:
  ```
  dev = qml.device("lightning.gpu", wires=6, batch_obs=True)
  ```
```

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -1,12 +1,38 @@
...
+  ```python
+  dev = qml.device("lightning.gpu", wires=6, batch_obs=True)
+  ...
+  @qml.qnode(dev, diff_method="adjoint")
+  def circuit(params):
+    for idx,w in enumerate(dev.wires):
+      qml.RX(params[idx], w)
+    return [qml.expval(qml.PauliZ(i))  for i in range(dev.num_wires)]
+  ```
+
```
Comment (mlxd) [inline .github/CHANGELOG.md:18]: For comparison, we can re-examine the benchmark script from the [Lightning GPU PennyLane blog post](https://pennylane.ai/blog/2022/07/lightning-fast-simulations-with-pennylane-and-the-nvidia-cuquantum-sdk/). Comparing with and without the multi-GPU supports on a machine with 4 A100 40GB GPUs shows a significant improvement over the single GPU run-times.

image

Comment (mlxd) [inline .github/CHANGELOG.md:18]:
```suggestion
For comparison, we can re-examine the benchmark script from the [Lightning GPU PennyLane blog post](https://pennylane.ai/blog/2022/07/lightning-fast-simulations-with-pennylane-and-the-nvidia-cuquantum-sdk/). Comparing with and without the multi-GPU supports on a machine with 4 A100 40GB GPUs shows a significant improvement over the single GPU run-times.

![image](https://user-images.githubusercontent.com/858615/184025758-7adeb433-5f7b-451a-bc72-ee3f7e321c49.png)

```

File: pennylane_lightning_gpu/_version.py [hunk 3]
```diff
@@ -16,4 +16,4 @@
    Version number (major.minor.patch[-label])
 """
 
-__version__ = "0.25.0-dev"
+__version__ = "0.25.0"
```
Comment (AmintorDusko) [inline pennylane_lightning_gpu/_version.py:19]:
```suggestion
__version__ = "0.25.0-rc0"
```

**Comment by AmintorDusko** [comment]
Looks good!

**Pull request merged**